R/preview-standalone.R

```diff
@@ -436,3 +436,83 @@ st_image_show <- function(path,
   )
   return(ans)
 }
+
+#' Render and save a table in png or pdf format
+#'
+#' @param x an stable object.
+#' @param file the final output file name; if `file` has `.tex` extension, the
+#' extension will be replaced with the value in `format`.
+#' @param stem used to form the final output file name; this argument will
+#' override what is provided in `file`.
+#' @param format the output file format.
+#' @param dir the final output directory.
+#' @param build_dir the directory where the image will be built.
+#' @param ... passed to [st_aspdf()] or [st_aspng()], depending on the value of
+#' `format`.
+#'
+#' @details
+#' The table will be built / rendered in `tempdir()`.
+#'
+#' @examples
+#' tab <- stable(stdata(), output_file = "foo.tex")
+#'
+#' \dontrun{
+#'   ans <- stable_save_image(tab format = "pdf")
+#'   basename(ans)
+#'   st_image_show(ans)
+#' }
+#'
+#' @seealso [st_as_image()]
+#'
+#' @export
+stable_save_image <- function(x,
+                              file = attr(x, "stable_file"),
+                              stem = NULL,
+                              format = c("png", "pdf"),
+                              dir = getOption("pmtables.dir"),
+                              build_dir = tempdir(), ...) {
+
+  if(!inherits(x, "stable")) {
+    stop(
+      "x is not an 'stable' object; ",
+      "maybe this object was corrupted or it wasn't generated from 'stable()'",
+      call.=FALSE
+    )
+  }
+  format <- match.arg(format)
+  if(is.character(file)) {
+    ext <- file_ext(file)
+    if(ext=="tex") {
+      stem <- file_path_sans_ext(file)
+      file <- paste0(stem, format)
```

Comment:
-   For "foo.tex" and `format = "pdf"`, this would incorrectly produce "foopdf".

-   That ends up not mattering because, in this `ext=="tex"` case, `stem` is a character, so the `file <- paste0(stem, ".", format)` on line 491 overrides this value.

So it seems line 487 can be dropped.

Reply:
Thanks; I noticed something wasn't behaving as I had expected when showing @KatherineKayMRG the code yesterday. I think this is the cause. I'll refactor this and confirm the behavior.